vissl/engines/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,545 @@
+        self.slurm_evaluator_checkpoint = slurm_evaluator_checkpoint
+        self.retry_evaluation_job_ids = retry_evaluation_job_ids or []
+        self.auto_retry_evaluations = auto_retry_evaluations
+        self.max_retries = max_retries
+
+        self._validate_class_instance()
+
+        # Will be set in #evaluate, once training_checkpoint_dir becomes available.
+        self.training_config = None
+        self.evaluation_results = None
+
+    def evaluate(self):
+        """
+        Evaluate the checkpoints. At a high level, this is the structure.
+
+        1. Load training YAML config file.
+        2. Monitor training checkpoints. When checkpoint is ready, launch the evaluation jobs.
+        3. Monitor the evaluation jobs. When evaluation jobs are complete, mark the results.
+        3. Once all jobs have been recorded, complete.
+        """
+        start_time = time.time()
+
+        # Wait for the training checkpoint folder to be available.
+        while True:
+            if time.time() - start_time > _TRAINING_CHECKPOINT_FILE_WAIT_SECONDS:
+                raise RuntimeError(
+                    f"Training checkpoint folder still doesn't exist after:"
+                    f"{_TRAINING_CHECKPOINT_FILE_WAIT_SECONDS / 60} minutes"
+                )
+
+            if PathManager.exists(self.training_checkpoint_file):
+                # Load training yaml config.
+                self._load_training_config()
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+        self.save_evaluation_results()
+
+        # Checkpoint folder is now available. Continuously monitor the training checkpoints,
+        # launch evaluation jobs as needed, monitor their progress, and record their results.
+        while True:
+            self._evaluate_checkpoints()
+
+            self._check_evaluation_jobs()
+
+            # Break if no more checkpoints to evaluate
+            if self._finished():
+                print("Evaluations are finished")
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+    def save_evaluation_results(self):
+        """
+        Upload json file to the evaluation directory.
+        """
+        # Upload all checkpoints evaluations to parent checkpoint directory.
+        evaluation_dir = self.evaluation_dir()
+        parent_metrics_file = os.path.join(evaluation_dir, "evaluation_metrics.json")
+        Path(evaluation_dir).mkdir(parents=True, exist_ok=True)
+        self._write_json_file(self.evaluation_results, parent_metrics_file)
+
+        # Upload each checkpoint's evaluations to child directories.
+        for checkpoint_str, benchmarks in self.evaluation_results.items():
+            child_metrics_dir = os.path.join(evaluation_dir, checkpoint_str)
+            child_metrics_file = os.path.join(
+                child_metrics_dir, "evaluation_metrics.json"
+            )
+            Path(child_metrics_dir).mkdir(parents=True, exist_ok=True)
```

Comment:
hmm we can try to use the `PathManager` ? the `Path` needs to be checked / tested if it will work with manifold (assuming we want to reuse this scheduler scripts more generically besides slurm)